logsapi/server_test.go

```diff
@@ -213,3 +212,13 @@ func TestTimestampsFunctionMessageWithDuration(t *testing.T) {
 	assert.Equal(t, ts.String(), event.Timestamp.String())
 
 }
+
+func TestLibhoneyEventWithSampleRate(t *testing.T) {
+	events := postMessages(t, []LogMessage{{
+		Time:   epochTimestamp,
+		Type:   "function",
+		Record: `{"time": "2020-12-25T12:34:56.789Z", "samplerate": 5, "data": {"foo": "bar", "duration_ms": 54} }`,
```

Comment:
Can you add test cases for sample rate being a:
- float
- string
- an invalid type like boolean